enos/modules/install_binaries/main.tf

```diff
@@ -0,0 +1,26 @@
+# Copyright (c) HashiCorp, Inc.
+# SPDX-License-Identifier: BUSL-1.1
+
+terraform {
+  required_providers {
+    enos = {
+      source = "registry.terraform.io/hashicorp-forge/enos"
+    }
+  }
+}
+
+module fetch_artifact {
+  for_each = toset(distinct(var.oss))
+  source   = "../fetch_artifactory"
+
+  artifactory_username = var.artifactory_username
+  artifactory_token    = var.artifactory_token
+  artifactory_host     = var.artifactory_host
+  artifactory_repo     = var.artifactory_repo
+  edition              = var.edition
+  os                   = each.value
+  product_version      = var.product_version
+  arch                 = var.arch
+  download_binary      = var.download_binaries
+  download_binary_path = "${var.download_binaries_path}/${each.value}"
+}
```

Comment:
Calling this module `install_binaries` is really confusing me. If `download_binary = true`, it only downloads the binaries locally and doesn't install them anywhere. If `download_binaries = false`, then as far as I can tell the only thing it does is expose the outputs from `data.eno_artifactory_item.nomad`.

It seems like the only reason this module exists is to handle the `for_each` on `var.oss`. Maybe we can push that down into `fetch_artifactory` instead and avoid having to come up with a name here?

Reply:
the logic to get the correct url for each binary is not trivial, and it will probably grow, I think we can benefit from having it in its own module

Reply:
Ok, no problem. let's merge this and we can iterate as we extend this.